python/lsst/obs/base/ingest.py

```diff
@@ -193,6 +195,9 @@ def __init__(self, config: Optional[RawIngestConfig] = None, *, butler: Butler,
         self.universe = self.butler.registry.dimensions
         self.datasetType = self.getDatasetType()
 
+        # Files that could not have metadata extracted
+        self._failed_metadata_files = []
+
```

Comment:
I had previously tried to avoid having any instance state that was affected by `run`, and while that's probably not a big deal here (it certainly isn't as long as we're using the command-line driver), it's definitely not a pattern I'd want other Tasks to copy.  Could we just have `prep` create and return this list to `run` instead?

Reply:
I had noticed the problem with rerunning ingest from the same object but `prep` does clear it each time so I don't think it's a real problem. I can take a look at returning the information instead though.